src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/config/WebClientConfig.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsrestrictedpatientsapi.config
+
+import UserContext
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpHeaders
+import org.springframework.web.reactive.function.client.ClientRequest
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeFunction
+import org.springframework.web.reactive.function.client.WebClient
+
+@Configuration
+class WebClientConfig(@Value("\${prison.api.endpoint.url}") private val prisonApiUrl: String) {
+
+  @Bean
+  fun prisonAPiWebClientAuditable(builder: WebClient.Builder): WebClient = builder
+    .baseUrl("$prisonApiUrl/api")
+    .filter(addAuthHeaderFilterFunction())
+    .build()
+
+  @Bean
+  fun prisonApiNoiAuthToken(builder: WebClient.Builder): WebClient = builder
+    .baseUrl("$prisonApiUrl")
+    .build()
+
+  private fun addAuthHeaderFilterFunction(): ExchangeFilterFunction =
+    ExchangeFilterFunction { request: ClientRequest?, next: ExchangeFunction ->
+      val filtered = ClientRequest.from(request)
+        .header(HttpHeaders.AUTHORIZATION, UserContext.getAuthToken())
```

Comment:
Potentially this could be derived straight from using `SecurityContextHolder.getContext().authentication as AuthAwareTokenConverter`.